cli/clistat/stat.go

```diff
@@ -0,0 +1,191 @@
+package clistat
+
+import (
+	"fmt"
+	"math"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/dustin/go-humanize"
+	"github.com/elastic/go-sysinfo"
+	"github.com/spf13/afero"
+	"golang.org/x/xerrors"
+	"tailscale.com/types/ptr"
+
+	sysinfotypes "github.com/elastic/go-sysinfo/types"
+)
+
+// Result is a generic result type for a statistic.
+// Total is the total amount of the resource available.
+// It is nil if the resource is not a finite quantity.
+// Unit is the unit of the resource.
+// Used is the amount of the resource used.
+type Result struct {
+	Total *float64 `json:"total"`
+	Unit  string   `json:"unit"`
+	Used  float64  `json:"used"`
+}
+
+// String returns a human-readable representation of the result.
+func (r *Result) String() string {
+	if r == nil {
+		return "-"
+	}
+
+	var usedDisplay, totalDisplay string
+	var usedScaled, totalScaled float64
+	var usedPrefix, totalPrefix string
+	usedScaled, usedPrefix = humanize.ComputeSI(r.Used)
+	usedDisplay = humanizeFloat(usedScaled)
+	if r.Total != (*float64)(nil) {
+		totalScaled, totalPrefix = humanize.ComputeSI(*r.Total)
+		totalDisplay = humanizeFloat(totalScaled)
+	}
+
+	var sb strings.Builder
+	_, _ = sb.WriteString(usedDisplay)
+
+	// If the unit prefixes of the used and total values are different,
+	// display the used value's prefix to avoid confusion.
+	if usedPrefix != totalPrefix || totalDisplay == "" {
+		_, _ = sb.WriteString(" ")
+		_, _ = sb.WriteString(usedPrefix)
+		_, _ = sb.WriteString(r.Unit)
+	}
```

Comment:
review: this means you will see output like `16.7 mcores/64 cores (0%)` or `1 B/100 TB (0%)`. I investigated scaling to use the same units, but this was non-trivial.